### PR TITLE
Roc-6450: Showing expert report direction permission in ccd ui

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6450

### Change description ###

Roc-6450: Showing expert report direction permission in ccd ui

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java [hunk 1]
```diff
@@ -46,6 +46,14 @@ public DirectionOrder from(CCDDirectionOrder ccdDirectionOrder, CCDOrderGenerati
         Optional.ofNullable(directionOrderData.getPaperDetermination()).ifPresent(paperDetermination ->
             builder.paperDetermination(YesNoOption.valueOf(paperDetermination.name())));
 
+        Optional.ofNullable(directionOrderData.getExpertReportPermissionPartyGivenToClaimant())
+            .ifPresent(permission -> builder
+                .expertReportPermissionGivenToClaimant(YesNoOption.valueOf(permission.name())));
+
+        Optional.ofNullable(directionOrderData.getExpertReportPermissionPartyGivenToDefendant())
+            .ifPresent(permission -> builder
+                .expertReportPermissionGivenToDefendant(YesNoOption.valueOf(permission.name())));
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: Please add these in assertion and unit test.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java [hunk 2]
```diff
@@ -55,6 +63,10 @@ public DirectionOrder from(CCDDirectionOrder ccdDirectionOrder, CCDOrderGenerati
             .extraDocUploadList(directionOrderData.getExtraDocUploadList().stream()
                 .map(CCDCollectionElement::getValue)
                 .collect(Collectors.toList()))
+            .expertReportInstructionsForClaimant(directionOrderData.getExpertReportInstructionClaimant()
+                .stream().map(CCDCollectionElement::getValue).collect(Collectors.toList()))
+            .expertReportInstructionsForDefendant(directionOrderData.getExpertReportInstructionDefendant()
+                .stream().map(CCDCollectionElement::getValue).collect(Collectors.toList()))
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: Please add these in assertion and unit test.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+@Component
+public class GenerateOrderRule {
+    public static final String CLAIMANT_REQUESTED_FOR_EXPORT_REPORT =
+        "Enter if you  grant permission for expert to the claimant";
+
+    public static final String DEFENDANT_REQUESTED_FOR_EXPORT_REPORT =
+        "Enter if you  grant permission for expert to the defendant";
+
+    public List<String> validateExpectedFieldsAreSelectedByLegalAdvisor(CCDCase ccdCase) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java]: missing unit tests for this method

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java (hunk 1 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: added

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java (hunk 2 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: added

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java (hunk 3 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java]: added

**Pull request merged**